### PR TITLE
[enhancement] add param which contain column comments of source table field

### LINKED ISSUE #3007: [enhancement] [StarRocks] When synchronizing tables using Flink CDC 3.x, is it possible to include the comments of the source table's fields when creating the target table?
### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Motivation

When synchronizing tables using Flink CDC 3.x, is it possible to include the comments of the source table's fields when creating the target table?

### Solution

_No response_

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by dufeng1010**
add param which contain column comments of source table field.
fix #3007

### CODE REVIEW

**Comment by dufeng1010** [comment]
@lvyanquan     Could you please help me review this PR

**Comment by everhopingandwaiting** [comment]
thanks for your implements

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/source/MySqlDataSourceOptions.java [hunk 1]
```diff
@@ -230,4 +230,12 @@ public class MySqlDataSourceOptions {
                     .defaultValue(true)
                     .withDescription(
                             "Whether send schema change events, by default is true. If set to false, the schema changes will not be sent.");
+
+    @Experimental
+    public static final ConfigOption<Boolean> ENABLE_COLUMN_COMMENTS =
+            ConfigOptions.key("enable.column.comments")
```
Comment (lvyanquan) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/source/MySqlDataSourceOptions.java:236]: Would it be better to use "column-comments.enabled"?

**Comment by everhopingandwaiting** [comment]
> In my testing, I found that when modifying both the field type and field comments of a table simultaneously, the comment information is not synchronized to the target database.


### test case


```sql

CREATE TABLE IF NOT EXISTS test_flinkcdc1
(
    XFID             varchar(64)                          not null comment 'Primary key ID'
        primary key,
    CXYY             text                                 null,
    CXSJ             datetime                             null comment 'Cancellation time',
    CREATE_TIME      timestamp  default CURRENT_TIMESTAMP null comment 'Creation time',
    UPDATE_TIME      timestamp  default CURRENT_TIMESTAMP null comment 'Modification time',
    CREATE_USER_ID   varchar(64)                          null comment 'Creator ID',
    CREATE_USER_NAME varchar(20)                          null comment 'Creator name',
    CREATE_ORG_CODE  varchar(64)                          null comment 'Creator organization code',
    CREATE_ORG_NAME  varchar(255)                         null comment 'Creator organization name',
    UPDATE_USER_ID   varchar(64)                          null comment 'Updater ID',
    UPDATE_USER_NAME varchar(20)                          null comment 'Updater name',
    UPDATE_ORG_CODE  varchar(64)                          null comment 'Updater organization code',
    UPDATE_ORG_NAME  varchar(255)                         null comment 'Updater organization name',
    DELETED          varchar(2)                           null comment 'Deletion flag; 0-Normal, 1-Deleted'
)
    comment 'Test Flink CDC 3 table' collate = utf8mb4_general_ci;

ALTER TABLE test_flinkcdc1 ADD COLUMN test11 text comment 'Test added field sync';

ALTER TABLE test_flinkcdc1 ADD COLUMN test12 text comment 'Test added field sync 2';
ALTER TABLE test_flinkcdc1 MODIFY COLUMN test12 text comment 'Test added field sync 2 modified type';

ALTER TABLE test_flinkcdc1 ADD COLUMN test13 text comment 'Test added field sync 3';

###

-- targetdb : auto-generated definition
CREATE TABLE test_flinkcdc1
(
    XFID             varchar(192)     not null comment 'Primary key ID',
    CXYY             varchar(1048576) null,
    CXSJ             datetime         null comment 'Cancellation time',
    CREATE_TIME      datetime         null comment 'Creation time',
    UPDATE_TIME      datetime         null comment 'Modification time',
    CREATE_USER_ID   varchar(192)     null comment 'Creator ID',
    CREATE_USER_NAME varchar(60)      null comment 'Creator name',
    CREATE_ORG_CODE  varchar(192)     null comment 'Creator organization code',
    CREATE_ORG_NAME  varchar(765)     null comment 'Creator organization name',
    UPDATE_USER_ID   varchar(192)     null comment 'Updater ID',
    UPDATE_USER_NAME varchar(60)      null comment 'Updater name',
    UPDATE_ORG_CODE  varchar(192)     null comment 'Updater organization code',
    UPDATE_ORG_NAME  varchar(765)     null comment 'Updater organization name',
    DELETED          varchar(6)       null comment 'Deletion flag; 0-Normal, 1-Deleted',
    test11           varchar(30)      null comment 'Test added field sync',
    test12           varchar(30)      null comment 'Test added field sync 2',
    test13           varchar(1048576) null comment 'Test added field sync 3'
)
    comment 'OLAP' engine = StarRocks;

```


error in `ALTER TABLE test_flinkcdc1 MODIFY COLUMN test12 text comment 'Test added field sync 2 modified type';`

**Comment by everhopingandwaiting** [comment]
For example, it was observed that the comment for the table name in the target database is fixed to "OLAP." Is it possible to also synchronize the table comment from the source table?

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/source/MySqlDataSourceOptions.java (hunk 1 above)
Comment (dufeng1010) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/source/MySqlDataSourceOptions.java:236]: ok, i fix it.

**Comment by dufeng1010** [comment]
> For starRocks db, it was observed that the comment for the table name in the target database is fixed to "OLAP." Is it possible to also synchronize the table comment from the source table?

@everhopingandwaiting i try to fix it.

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/schema/Schema.java
Comment (everhopingandwaiting) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/schema/Schema.java:1]: nice work

**Comment by dufeng1010** [comment]
@lvyanquan @everhopingandwaiting  Compilation failure is not related to the submitted code

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReaderTest.java [hunk 2]
```diff
@@ -199,7 +199,7 @@ public void testFinishedUnackedSplitsUsingStateFromSnapshotPhase() throws Except
                     "+I[2000, user_21, Shanghai, 123567891234]"
                 };
         // Step 2: wait the snapshot splits finished reading
-        Thread.sleep(5000L);
```
[the remark this answers is not in the document]
Comment (dufeng1010) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReaderTest.java:202]: sometimes 5 seconds cann't finish read, so i increment sleep time to 10 second.

**Comment by lvyanquan** [comment]
Hi @dufeng1010  can you help to rebase to master？

**Pull request closed without merging**